### PR TITLE
Parallel

### PR DESCRIPTION
**Pull request by otaviobp**
Fixes in parallel download and workaround to avoid bug #562.

@gtkramer: This PR fixes the bug you reported. Can you check if this version is working for you?

### CODE REVIEW

File: src/verify.c [hunk 1]
```diff
@@ -118,6 +119,7 @@ static void print_help(const char *name)
 	fprintf(stderr, "   -t, --time              Show verbose time output for swupd operations\n");
 	fprintf(stderr, "   -N, --no-scripts        Do not run the post-update scripts and boot update tool\n");
 	fprintf(stderr, "   -b, --no-boot-update    Do not install boot files to the boot partition (containers)\n");
+	fprintf(stderr, "   -D, --max-parallel-pack-downloads =[n] Set the maximum number of parallel pack downloads in parallel\n");
```
Comment (matthewrsj) [inline src/verify.c]: Why the space between `--max-parallel-pack-downloads` and `=`?

File: src/update.c [hunk 2]
```diff
@@ -625,6 +626,7 @@ static void print_help(const char *name)
 	fprintf(stderr, "   -t, --time              Show verbose time output for swupd operations\n");
 	fprintf(stderr, "   -N, --no-scripts        Do not run the post-update scripts and boot update tool\n");
 	fprintf(stderr, "   -b, --no-boot-update    Do not update the boot files using clr-boot-manager\n");
+	fprintf(stderr, "   -D, --max-parallel-pack-downloads =[n] Set the maximum number of parallel pack downloads in parallel\n");
```
Comment (matthewrsj) [inline src/update.c]: the space question again.

**Comment by otaviobp** [comment]
v2: 
 - Removed whitespaces after '='
 - Added comment on MAX_XFER change

File: src/packs.c [hunk 3]
```diff
@@ -36,8 +36,14 @@
...
 
 /* hysteresis thresholds */
-#define MAX_XFER 25
-#define MAX_XFER_BOTTOM 15
+/* TODO: update MAX_XFER after bug #562 is fixed.
+ *
+ * MAX XFER is the number of simultaneous downloads to be performed at the same
+ * time. This value is set to 1 because of a bug that causes a dowload problem
+ * when extrating a large file while other is being downloaded. Set this value
+ * to a larger number, to be defined by tests, after bug is fixed.
```
Comment (matthewrsj) [inline src/packs.c]: Thank you.

File: src/clr_bundle_add.c [hunk 4]
```diff
@@ -57,6 +57,7 @@ static void print_help(const char *name)
 	fprintf(stderr, "   -t, --time              Show verbose time output for swupd operations\n");
 	fprintf(stderr, "   -N, --no-scripts        Do not run the post-update scripts and boot update tool\n");
 	fprintf(stderr, "   -b, --no-boot-update    Do not update the boot files using clr-boot-manager\n");
+	fprintf(stderr, "   -D, --max-parallel-pack-downloads=[n] Set the maximum number of parallel pack downloads in parallel\n");
```
Comment (matthewrsj) [inline src/clr_bundle_add.c]: This should probably *also* be a compile-time option, as well as have this flag added to the readme. I want to get this merged though so it doesn't have to be done for this PR.

Comment (otaviobp) [reply src/clr_bundle_add.c]: Added that to the Readme. We can let this to be a compile-time option in a new PR.

**Pull request merged**